packages/sdk/src/hooks/use-async-dispatch.ts

```diff
@@ -1,13 +1,16 @@
 import type { Dispatch } from 'redux';
+import { useDispatch } from 'react-redux';
 import type { TSdkAction } from '../types';
 
-import { useDispatch } from 'react-redux';
+type TAsyncDispatch<Action extends TSdkAction, Data> = (
+  action: Action
+) => Promise<Data>;
 
 // Wraps `dispatch` and cast the return type to a Promise, as the middleware
 // returns a Promise.
 function useAsyncDispatch<Action extends TSdkAction, Data>() {
   const dispatch = useDispatch<Dispatch<Action>>();
-  return (action: Action) => dispatch(action) as unknown as Promise<Data>;
```

Comment:
I think this function was only used for defining the type casting, as it doesn't do anything specific.

We can remove it and simply return the `dispatch` function and to the type casting differently.